internals/secrethub/account_init.go

```diff
@@ -158,36 +160,45 @@ func (cmd *AccountInitCommand) Run() error {
 			credentialPath,
 		)
 
+		credential := credentials.CreateKey()
+
 		// Only prompt for a passphrase when the user hasn't used --force.
 		// Otherwise, we assume the passphrase was intentionally not
 		// configured to output a plaintext credential.
+		var passphrase string
 		if !cmd.credentialStore.IsPassphraseSet() && !cmd.force {
-			passphrase, err := ui.AskPassphrase(cmd.io, "Please enter a passphrase to protect your local credential (leave empty for no passphrase): ", "Enter the same passphrase again: ", 3)
+			var err error
+			passphrase, err = ui.AskPassphrase(cmd.io, "Please enter a passphrase to protect your local credential (leave empty for no passphrase): ", "Enter the same passphrase again: ", 3)
 			if err != nil {
 				return err
 			}
-			cmd.credentialStore.SetPassphrase(passphrase)
 		}
 
 		fmt.Fprint(cmd.io.Stdout(), "Generating credential...")
-		credential, err := secrethub.GenerateCredential()
+		err := credential.Create()
+
+		fmt.Fprintln(cmd.io.Stdout(), " Done")
+
+		exportKey := credential.Key
+		if passphrase != "" {
+			exportKey = exportKey.Passphrase(credentials.FromString(passphrase))
+		}
+
+		exportedCredential, err := exportKey.Export()
 		if err != nil {
 			return err
 		}
-		cmd.credentialStore.Set(credential)
-		fmt.Fprintln(cmd.io.Stdout(), " Done")
-
-		err = cmd.credentialStore.Save()
+		err = cmd.credentialStore.ConfigDir().Credential().Write(exportedCredential)
 		if err != nil {
 			return err
 		}
 
-		verifier, err := credential.Verifier()
+		verifierBytes, err := exportKey.Verifier().Verifier()
 		if err != nil {
 			return err
 		}
 
-		fingerprint, err := credential.Fingerprint()
+		fingerprint, err := api.GetFingerprint(exportKey.Verifier().Type(), verifierBytes)
```

Comment:
Discussed this in person. Ideally, we'd like to be able to call `Fingerprint` on a verifier.

Reply:
https://github.com/secrethub/secrethub-go/pull/118